### PR TITLE
Rebase update-alternatives check for new rpmlint

### PR DESCRIPTION
**Pull request by scarabeusiv**
Followup of #369 as it should be distro agnostic.

Still needs tests before merging and actually some code cleanup&checking.

### CODE REVIEW

File: rpmlint/checks/UpdateAlternativesCheck.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import rpm
+from rpmlint.checks.AbstractCheck import AbstractCheck
+from rpmlint.pkg import b2s
+
+
+class UpdateAlternativesCheck(AbstractCheck):
+    """
+    Check for compliance with SUSE's update-alternatives guidelines:
+      http://en.opensuse.org/openSUSE:Packaging_Multiple_Version_guidelines
```
Comment (Conan-Kudo) [inline rpmlint/checks/UpdateAlternativesCheck.py]: Can we make this comment more generic?

File: rpmlint/descriptions/UpdateAlternativesCheck.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+not marked as ghost. Mark it as %ghost.
+"""
+alternative-link-missing="""
+The file %{_sysconfdir}/alternatives/$(basename generic-name) is missing
+in the file list. Mark it as %ghost and add it to the file list.
+"""
+suse-alternative-generic-name-missing="""
+The update-alternatives generic name is not in the filelist. Create it as
+a symlink to %{_sysconfdir}/alternatives/$(basename generic-name) and add it
+to the file list.
```
Comment (Conan-Kudo) [inline rpmlint/descriptions/UpdateAlternativesCheck.toml]: We should drop the SUSE prefix here...

**Comment by Conan-Kudo** [comment]
@scarabeusiv Can we squash these two into a single commit?

**Comment by scarabeusiv** [comment]
I plan to squash it at the end. So i could see what i messed up and what was there from the begining.

**Comment by Conan-Kudo** [comment]
@scarabeusiv Makes sense to me. 👍

File: rpmlint/descriptions/UpdateAlternativesCheck.toml (hunk 2 above)
Comment (scarabeusiv) [reply rpmlint/descriptions/UpdateAlternativesCheck.toml]: done, copy&pasto.

File: rpmlint/checks/UpdateAlternativesCheck.py (hunk 1 above)
Comment (scarabeusiv) [reply rpmlint/checks/UpdateAlternativesCheck.py]: Adjusted it somehow, still i think we should rather point to the docs than describe it again in here? Or should i copy the wiki content there?

Comment (Conan-Kudo) [reply rpmlint/checks/UpdateAlternativesCheck.py]: If it's not too much content, perhaps inlining it makes sense? I'm not sure.

**Comment by scarabeusiv** [comment]
After I wrote down what the checker should do it seems there is quite huge room for improvement to actually have it do what it is supposed...

File: rpmlint/checks/UpdateAlternativesCheck.py (hunk 1 above)
Comment (scarabeusiv) [reply rpmlint/checks/UpdateAlternativesCheck.py]: added there.

**Comment by Conan-Kudo** [comment]
@scarabeusiv Let's rename this to `AlternativesCheck`, as that's a better name for this.

**Comment by scarabeusiv** [comment]
@Conan-Kudo okay.

**Comment by scarabeusiv** [comment]
And in we go. I should probably mark the leap build as expected fail as it will take us few weeks to get that in.

**Pull request merged**